tests/functional/base_test.go

```diff
@@ -52,6 +52,33 @@ func DefaultDataPlaneSpec() dataplanev1.OpenStackDataPlaneSpec {
 	}
 }
 
+func DataPlaneWithNodeSpec() dataplanev1.OpenStackDataPlaneSpec {
+	return dataplanev1.OpenStackDataPlaneSpec{
+		DeployStrategy: dataplanev1.DeployStrategySection{
+			Deploy: false,
+		},
+		Roles: map[string]dataplanev1.OpenStackDataPlaneRoleSpec{
+			"edpm-compute": {
+				Services: []string{"configure-network", "validate-network", "install-os", "configure-os", "run-os"},
+				NodeTemplate: dataplanev1.NodeSection{
+					AnsibleSSHPrivateKeySecret: "dataplane-ansible-ssh-private-key-secret",
+				},
+			},
+		},
+		Nodes: map[string]dataplanev1.OpenStackDataPlaneNodeSpec{
+			"edpm-compute-0": {
+				Node: dataplanev1.NodeSection{
+					NetworkConfig: dataplanev1.NetworkConfigSection{
+						Template: "{'test-nic-override-value'}",
```

Comment:
nit i would drop the {} it could be confuseed with jinja which it is not
its also not a dict or json object.

Template: "{'test-nic-override-value'}", -> Template: "'test-nic-override-value'",

Reply:
perhasp useing a subset of the sample config woudl be better

```
---
            network_config:
            - type: interface
              name: nic2
              mtu: 1500
              addresses:
                - ip_netmask:
                    {{ ctlplane_ip }}/{{ ctlplane_subnet_cidr }}
```
something like this
```
Node: dataplanev1.NodeSection{
	NetworkConfig: dataplanev1.NetworkConfigSection{
		Template: 
			`---
			network_config:
		  	  - type: interface
		            name: nic2
		            mtu: 1500
              		    addresses:
                	      - ip_netmask:
                    	        {{ ctlplane_ip }}/{{ ctlplane_subnet_cidr }}
			`,
```

Reply:
actully pulling it out into a var would be better then inline if your using raw stings
i think all the indentation would be preserved in the finale output but i haven't
really used them.

```
template := `---
network_config:
  - type: interface
    name: nic2
    mtu: 1500
    addresses:
      - ip_netmask:
          {{ ctlplane_ip }}/{{ ctlplane_subnet_cidr }}
`
Node: dataplanev1.NodeSection{
	NetworkConfig: dataplanev1.NetworkConfigSection{
		Template: template
...

```